likecoin/js/admin/likecoin_metabox.js

```diff
@@ -56,6 +56,58 @@ async function onISCNCallback(event) {
   ISCNTextField.innerHTML = `<a rel="noopener" target="_blank" href="${url}">${status}</a>`;
 }
 
+async function onLikePayCallbackAndUploadArweave(event) {
+  if (event.origin !== 'https://like.co') { // For development, skip this line.
+    return;
+  }
+  try {
+    const { action, data } = JSON.parse(event.data);
+    if (action !== 'TX_FEE_SUBMITTED') return;
+    const { tx_hash: txHash, error, success } = data;
+    if (error || success === false) return;
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/arweave/upload`,
+      dataType: 'json',
+      contentType: 'application/json; charset=UTF-8',
+      data: JSON.stringify({ txHash }),
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    if (!res.data) {
+      window.alert('Something went wrong when uploading to Arweave. Please re-submit.');
+      return;
+    }
+    if (res.data.arweaveId && res.data.ipfsHash) {
+      submitToISCNAfterArweaveUpload(res.data);
+    }
+  } catch (error) {
+    console.error(`Error occurs when uploading to Arweave: ${error.message}`);
+  }
+}
+
+async function submitToISCNAfterArweaveUpload(arweaveInfo) {
+  const { title, tags, url } = lcPostInfo;
+  const { arweaveId } = arweaveInfo;
+  const { siteurl } = wpApiSettings;
+  // ipfsHash is from Wordpress DB. Could be matters-formatted or likecoin-formatted
+  const { ipfsHash } = arweaveInfo;
+  if (!arweaveId || !ipfsHash) return;
+  const titleString = encodeURIComponent(title);
+  const tagsArray = tags || [];
+  const tagsString = tagsArray.join(',');
+  const urlString = encodeURIComponent(url);
+  const redirectString = encodeURIComponent(siteurl);
+  const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${ipfsHash}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&url=${urlString}&redirect_uri=${redirectString}`;
+  window.open(
+    likeCoISCNWidget,
+    '_blank',
+    'menubar=no,location=no,width=576,height=768',
+  );
+  window.addEventListener('message', onISCNCallback, false);
+}
```

Comment:
Should reuse the current `onSubmitToISCN` flow instead of making a new function, but do make sure we support both arweave only and matters 
only upload flow